File: src/main/java/org/pixeltime/enchantmentsenhance/command/SubCommand.java
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.command;

import org.bukkit.entity.Player;

public abstract class SubCommand {

    /**
     * /<command> <subcommand> args[0] args[1]
     */
    public SubCommand() {
    }

    /**
     * Player oncommand.
     *
     * @param player
     * @param args
     */
    public abstract void onCommand(Player player, String[] args);

    /**
     * @return Command name.
     */
    public abstract String name();

    /**
     * @return Command information.
     */
    public abstract String info();

    /**
     * @return Command aliases.
     */
    public abstract String[] aliases();

    /**
     * @return Command permission.
     */
    public abstract String getPermission();
}

File: modules/EnchantmentsEnhance-Plugin/src/main/kotlin/org/pixeltime/enchantmentsenhance/command/SubCommand.kt
package org.pixeltime.enchantmentsenhance.command

import org.bukkit.entity.Player
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

/**
 * /<command> <subcommand> args[0] args[1]
</subcommand></command> */
abstract class SubCommand {

    /**
     * @return Command permission.
     */
    abstract val permission: String

    /**
     * Player oncommand.
     *
     * @param player
     * @param args
     */
    abstract fun onCommand(player: Player, args: Array<String>)

    /**
     * @return Command name.
     */
    abstract fun name(): String

    /**
     * @return Command information.
     */
    abstract fun usage(): String

    /**
     * @return Command aliases.
     */
    abstract fun aliases(): Array<String>

    fun info(): String {
        return "&b${usage()} ${SettingsManager.lang.getString("help.${name()}")}"
    }
}
